src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -29,22 +29,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     String userId = "null";
     String logoutUrl = "null";
     String loginUrl = "null";
+    String userEmail = "null";
 
     // Boolean to update User's main page 
-    String showNewTab = false;
+    Boolean showNewTab = false;
 
     if (userService.isUserLoggedIn()) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String userId = userService.getCurrentUser().getUserId();
-      String logoutUrl = userService.createLogoutURL(urlToRedirect);
-      String showNewTab = true;
+      userEmail = userService.getCurrentUser().getEmail();
+      userId = userService.getCurrentUser().getUserId();
+      logoutUrl = userService.createLogoutURL(urlToRedirect);
+      showNewTab = true;
     } else {
-      String loginUrl = userService.createLoginURL(urlToRedirect);
+      loginUrl = userService.createLoginURL(urlToRedirect);
     }
 
-    User user = new User(userId, loginUrl, logoutUrl, email, showNewTab);
+    User user = new User(userId, loginUrl, logoutUrl, userEmail, showNewTab);
```

Comment:
Why are we storing the login and logout URLs, and whether to show a new tab in the user object and datastore?

These are ephemeral values and don't need to be stored.

Reply:
Fixed in #5